feat: Added form control

albinAppsmith opened the pull request:
## Description

Added form control component to design system. This components comes with label, helper and error styles.

Fixes https://github.com/appsmithorg/appsmith/issues/27770

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Issues raised during DP testing


## Checklist:
### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes


### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

laveena-en commented:
@albinAppsmith 

Color of the helper text isn't as per the design file
`Helper` has been misspelled 
On disabling, only the helper text changes colour but the component remains active, is this expected?

albinAppsmith commented:
> @albinAppsmith
> 
> 1. Color of the helper text isn't as per the design file
> 2. `Helper` has been misspelled
> 3. On disabling, only the helper text changes colour but the component remains active, is this expected?

1. Fixed this
2. Fixed this
3. Yes, the component have a seperate prop to disable it. So when form control is disabled control will not get disabled.

laveena-en commented:
Tested and verified, PR looks good to me.
